vortex-expr/src/transform/immediate_access.rs

```diff
@@ -0,0 +1,84 @@
+use itertools::Itertools;
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_array::aliases::hash_set::HashSet;
+use vortex_dtype::{FieldName, StructDType};
+use vortex_error::VortexResult;
+
+use crate::traversal::{Node, NodeVisitor, TraversalOrder};
+use crate::{ExprRef, GetItem, Identity, Select};
+
+pub(crate) type FieldAccesses<'a> = HashMap<&'a ExprRef, HashSet<FieldName>>;
+
+/// For all subexpressions in an expression, find the fields that are accessed directly from the
+/// scope, but not any fields in those fields
+/// e.g. scope = {a: {b: .., c: ..}, d: ..}, expr = ident().a.b + ident().d accesses {a,d} (not b).
+pub(crate) struct ImmediateIdentityAccessesAnalysis<'a> {
```

Comment:
If we're cleaning up, can we start the convention of exporting a single function with the same name as the module, then keeping all structs internal? 

You may have to define a result type struct for complex result.

Reply:
Sure

Reply:
You should move the comment to the public function

Reply:
https://github.com/spiraldb/vortex/pull/2017